src/test/java/com/digitalpebble/spruce/modules/boavizta/BoaviztAPIClientTest.java

```diff
@@ -0,0 +1,373 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalpebble.spruce.modules.boavizta;
+
+import com.digitalpebble.spruce.Provider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class BoaviztAPIClientTest extends AbstractBoaviztaTest {
+
+    private BoaviztAPIClient client;
+    private static final String TEST_HOST = "http://localhost:5000";
+
+    @BeforeEach
+    void setUp() {
+        client = new BoaviztAPIClient(TEST_HOST);
+    }
+
+    @Test
+    void testConstructor() {
+        BoaviztAPIClient testClient = new BoaviztAPIClient("http://custom-host:8080");
+        assertNotNull(testClient);
+    }
+
+    @Test
+    void testConstructorWithNullHost() {
+        // The constructor doesn't actually throw NullPointerException for null host
+        // It just stores the null value, which will cause issues later when making HTTP calls
+        BoaviztAPIClient testClient = new BoaviztAPIClient(null);
```

Comment:
it would be worth changing the behaviour of the constructor so that an error is thrown